File: src/main/java/ch/uzh/ifi/access/model/CourseInformation.java
package ch.uzh.ifi.access.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class CourseInformation {
    @Id
    @GeneratedValue
    public Long id;

    @JsonIgnore
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "course_id")
    public Course course;

    @Column(nullable = false)
    public String language;

    @Column(nullable = false)
    public String title;

    @Column(nullable = false)
    public String description;

    @Column(nullable = false)
    public String university;

    @Column(nullable = false)
    public String period;
}

File: src/main/java/ch/uzh/ifi/access/model/CourseInformation.kt
package ch.uzh.ifi.access.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.Getter
import lombok.Setter

@Entity
class CourseInformation {
    @Id
    @GeneratedValue
    var id: Long? = null

    @JsonIgnore
    @ManyToOne(cascade = [CascadeType.ALL])
    @JoinColumn(name = "course_id")
    var course: Course? = null

    @Column(nullable = false)
    var language: String? = null

    @Column(nullable = false)
    var title: String? = null

    @Column(nullable = false)
    var description: String? = null

    @Column(nullable = false)
    var university: String? = null

    @Column(nullable = false)
    var period: String? = null
}
